Description: Do not trim the results of FcFontSort(),
 as this may miss some reasonable candidates under certain circumstances.
 BTW, fix passing a non-pointer as a pointer to "result".
Author: Fabian Greffrath <fabian+debian@greffrath.com>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=670055

--- cups-filters-1.0.16.orig/filter/texttopdf.c
+++ cups-filters-1.0.16/filter/texttopdf.c
@@ -51,6 +51,7 @@ EMB_PARAMS *font_load(const char *font)
   FcPattern *pattern;
   FcFontSet *candidates;
   FcChar8   *fontname = NULL;
+  FcResult   result;
   int i;
 
   if ( (font[0]=='/')||(font[0]=='.') ) {
@@ -64,7 +65,7 @@ EMB_PARAMS *font_load(const char *font)
     FcDefaultSubstitute (pattern);
 
     /* Receive a sorted list of fonts matching our pattern */
-    candidates = FcFontSort (0, pattern, FcTrue, 0, 0);
+    candidates = FcFontSort (0, pattern, FcFalse, 0, &result);
     FcPatternDestroy (pattern);
 
     /* In the list of fonts returned by FcFontSort()

